### DISCUSSION TITLE
Separate Scorecard maintainers meet

### DISCUSSION THREAD

**Discussion by azeemshaikh38** [ossf/scorecard#1723] [Ideas]
@naveensrinivasan @laurentsimon @justaugustus @jeffmendoza 

Hi folks,

Recently, we have been running short on time to discuss all issues on the bi-weekly agenda (which is awesome, cause that shows how much we have grown!). So some maintainers have suggested having a separate maintainers meet and I agree with that.

Proposing a separate bi-weekly meet for Scorecard/AllStar maintainers. We can use the community meeting to do high-level updates on upcoming features, goals, vision etc. so that we can inform our users about the changes that might be coming and also address any issues/concerns they might have about our tools. Can use the maintainers meet to do design reviews and technical discussions etc.

If that sounds good to folks, I'll try and setup a time that works for everyone.

**Comment by naveensrinivasan** [comment]
👍. Thanks @azeemshaikh38.

**Comment by azeemshaikh38** [comment]
Friendly ping - @laurentsimon @jeffmendoza @justaugustus

**Reply by laurentsimon** [reply]
fine with me.

**Comment by justaugustus** [comment]
@ossf/scorecard-admins -- I'll be the contrarian for a few reasons:

- extra meeting on the calendar
- high chance of making decision on directions for various efforts in a less-public space (we should bias towards high-transparency whenever/however possible)

That said, I'll use Kubernetes SIG Release as an example:
- SIG leads meet on Mondays to solidify agenda for the public meetings (and any potentially sensitive security-/HR-type issues)
- On Tuesdays, the SIG alternates between the SIG general meeting and the Release Engineering subproject meeting

In spaces where it makes sense, we use Slack and start a standup thread to make decisions/query for updates.
(We could consider doing this on GH discussions too!)

I think a few additional meetings could make sense to sift through the backlog/roadmap, but I'm not sure that they need to exist long-term (SIG Release leads moved our weekly sync to biweekly, because there just wasn't the need to sync that frequently in a private setting).

If we start doing maintainer syncs, I'd suggest holding them at the same time as the public meetings, just on alternating weeks.

**Comment by jeffmendoza** [comment]
I don't think we should have a "maintainers meet", as in we shouldn't have a specific meeting with a intended limited invite list.

If we are running out of time in our bi-weekly (I agree we are). I'd vote to change the sync to weekly, and have alternating topics by week

A. Strategy, feature proposal, scoring discussions
B. Work through backlog of specific open Issues and PRs only

I believe this is the intention here, but we shouldn't try to achieve B by making an exclusive meeting. Just setting the meeting agenda/topic ahead of time and sticking to it should achieve what we want I believe.

**Reply by laurentsimon** [reply]
I like the idea

**Reply by justaugustus** [reply]
> Just setting the meeting agenda/topic ahead of time and sticking to it should achieve what we want I believe.

This is my assessment as well. I think the problem that we're having is lack of time because we're jumping between a few different things instead of sticking to the agenda.

We've got a decent format, but I think what would help refine it is, "What's top of mind?" and by extension, "Are we providing updates that align with that?"
(We'll get better at this over time)

Last year, Kubernetes SIG Release produced a roadmap: https://github.com/kubernetes/sig-release/blob/master/roadmap.md
Which has been helpful to gut-check on how we should align/what we should be aligning on.

To date, my contributions have been around:
- things I have direct interest/use for
- patterns I've seen/used in the past that can be generally useful (contribution policy, raising visibility, code/effort deduplication)

**Reply by azeemshaikh38** [reply]
So my main reasoning to propose a separate meeting is because certain topics are more useful to maintainers rather than the larger community. The larger community will have little to actively contribute to these topics and over time might start finding these meets less beneficial to attend.

I strongly think there is an advantage in having a limited invitee meet so that the bi-weekly meet can cater to more community oriented updates about - "what's new in Scorecard", "are we solving problems that are inclusive of the community" etc. which require strong community participation.

**Reply by justaugustus** [reply]
> So my main reasoning to propose a separate meeting is because certain topics are more useful to maintainers rather than the larger community.

If this is the primary motivator, then I'll change my stance from "meh, okay, maybe" to "strong -1".

We're going to have scores of users/contributors dropping by with varying backgrounds.
By making an assumption about what is important to them, we do a few things:
- make unnecessary/inaccurate assumptions
- cultivate exclusion and silo information
- limit the opinions we're considering about "maintainer" topics
- limit contribution opportunities for potential maintainers

The only time I consider limited-scope/private spaces is when the conversation is sensitive and I'm not sure we've made the case here that that's true.

**Reply by azeemshaikh38** [reply]
Since the majority agrees with not having a separate meet, ok to close this discussion. We can continue to use the bi-weekly for maintainer discussions.

**Reply by laurentsimon** [reply]
I thought the idea was to make them weekly, rather than bi-weekly?
